test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,196 @@
+package field_selectors_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	deviceYAMLPath = "examples/device.yaml"
+	fleetYAMLPath  = "examples/fleet.yaml"
+)
+
+// Define the struct for parsing device.yaml
+type Device struct {
+	Metadata struct {
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestFieldSelectors(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Field Selectors E2E Suite")
+}
+
+var _ = Describe("Field Selectors in Flight Control", func() {
+
+	var (
+		harness *e2e.Harness
+		token   string
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+
+		loginArgs := []string{"login", "${API_ENDPOINT}", "--insecure-skip-tls-verify"}
```

Comment:
Maybe we could create a login function, since I believe we will re-use this in other tests